cmd/operator/main.go

```diff
@@ -81,11 +102,75 @@ func getIsWebhookEnabled() bool {
 	return enabled
 }
 
+// getEncoderConfig returns a concrete encoders configuration
+func getEncoderConfig(devMode bool) zapcore.EncoderConfig {
+	encoderConfig := zap.NewProductionEncoderConfig()
+	if devMode {
+		encoderConfig = zap.NewDevelopmentEncoderConfig()
+	}
+	return encoderConfig
+}
+
+// getLogWriter returns an io.writer (setting up rolling files) converted
+// into a zapcore.WriteSyncer
+func getLogWriter(path string, age, size, rotation int) zapcore.WriteSyncer {
+	lumberJackLogger := &lumberjack.Logger{
+		Filename:   path,
+		MaxSize:    size, // megabytes
+		MaxBackups: rotation,
+		MaxAge:     age, // days
+	}
+	return zapcore.AddSync(lumberJackLogger)
+}
+
+// getZapcoreLevel takes the level as string and returns the corresponding
+// zapcore.Level. If the string level is invalid, it returns the default
+// level
+func getZapcoreLevel(lvl string) zapcore.Level {
+	lvls := map[string]zapcore.Level{
+		"debug": zapcore.DebugLevel,
+		"info":  zapcore.InfoLevel,
+		"warn":  zapcore.WarnLevel,
+		"error": zapcore.ErrorLevel,
+	}
+	if _, found := lvls[lvl]; !found {
+		// returns default level if the input level is invalid
+		log.Println(fmt.Sprintf("Invalid level, %s level will be used instead", Level))
+		return lvls[Level]
+	}
+	return lvls[lvl]
+}
+
+// getLogger is a wrapper that calls other functions
+// to build a logger.
+func getLogger(logArgs Logging, devMode bool) *zap.Logger {
+	encoderConfig := getEncoderConfig(devMode)
+	writes := []zapcore.WriteSyncer{}
+	lvl := zap.NewAtomicLevelAt(zap.DebugLevel)
+	if logArgs.FilePath != "" {
+		w := getLogWriter(logArgs.FilePath, logArgs.MaxFileAge, logArgs.MaxFileSize, logArgs.MaxFileRotation)
```

Comment:
Can we just pass in the Logging struct instead of each individual field?

Reply:
Yes I am already changing that.